builder/amazon/chroot/step_attach_volume.go

```diff
@@ -87,10 +88,33 @@ func (s *StepAttachVolume) Run(state multistep.StateBag) multistep.StepAction {
 		return multistep.ActionHalt
 	}
 
+	err = statDevice(attachVolume, state)
+	if err != nil {
+		state.Put("error", err)
+		ui.Error(err.Error())
+		return multistep.ActionHalt
+	}
 	state.Put("attach_cleanup", s)
 	return multistep.ActionContinue
 }
 
+func statDevice(attachVolume string, state multistep.StateBag) error {
+	statAttempts := 0
+	deviceAltName := strings.Replace(attachVolume, "/sd", "/xvd", 1)
+	for statAttempts < 30 {
```

Comment:
Please use an [inline `for` loop](https://golang.org/doc/effective_go.html#for), e.g:

```
for i := 0; i < 10; i++ {
    // do stuff
}
```